Makefile

```diff
@@ -0,0 +1,66 @@
+PROJECT := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+
+SRC := $(PROJECT)src
+TESTS := $(PROJECT)tests
+ALL := $(SRC) $(TESTS)
+
+export PYTHONPATH = $(PROJECT):$(PROJECT)/lib:$(SRC)
+export PY_COLORS=1
+
+# Update uv.lock with the latest deps
+lock:
+	uv lock --upgrade --no-cache
+
+# Generate requirements.txt from pyproject.toml
+requirements:
+	uv export --frozen --no-hashes --format=requirements-txt -o requirements.txt
+
+# Lint the code
+lint:
+	uv run --isolated --extra lint \
```

Comment:
Switch to `uv run --frozen ...`

This will not update the `uv.lock` file. What is the difference between:

Note: They cannot be used together!
`--frozen`
- pyproject can differ from uv.lock and will still run the command but will not update the lock file.
`--locked`
- Needs to be exactly like uv.lock or nothing happens